### ISSUE TITLE
left `joinWith` not working unless all fields of the right case class are `Option`

### ISSUE THREAD

**Issue by joan38** [vincenzobaz/spark-scala3#46]
Are you guys able to make `joinWith` work in Scala 3?
I get the following error:
```
23/10/04 18:39:59 WARN TaskSetManager: Lost task 2.0 in stage 10.0 (TID 28203) (ip-100-74-17-116.ec2.internal executor 96): java.lang.NullPointerException
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$SpecificSafeProjection.NewInstance_1$(Unknown Source)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$SpecificSafeProjection.apply(Unknown Source)
	at scala.collection.Iterator$$anon$9.next(Iterator.scala:584)
	at scala.collection.Iterator$$anon$9.next(Iterator.scala:584)
	at scala.collection.Iterator$$anon$9.next(Iterator.scala:584)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage10.hashAgg_doAggregateWithKeys_0$(Unknown Source)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage10.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
	at scala.collection.Iterator$$anon$9.hasNext(Iterator.scala:583)
	at org.apache.spark.shuffle.sort.UnsafeShuffleWriter.write(UnsafeShuffleWriter.java:179)
	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
	at org.apache.spark.scheduler.Task.run(Task.scala:136)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:555)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1531)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:558)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
```

It sounds like the NPE is coming on executors.

Thanks

**Comment by joan38** [comment]
I just update the issue since I found out that the issue in coming from `joinWith`

**Comment by vincenzobaz** [comment]
It's very nice that we start seeing issues from real use cases :)
Thank you for opening this!

**Comment by joan38** [comment]
Thanks to YOU for this lib, enabling us to us Scala 3!
BTW we are using it @Netflix for processing a few Tb/hour of playback data 😉

**Comment by joan38** [comment]
Fixed by #45
